clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/security/KubernetesCredentials.java

```diff
@@ -18,20 +18,22 @@
 
 import io.fabric8.kubernetes.client.KubernetesClient;
 
+import java.util.List;
+
 public class KubernetesCredentials {
   private final KubernetesClient client;
-  private final String namespace;
+  private final List<String> namespaces;
 
-  public KubernetesCredentials(String namespace, KubernetesClient client) {
+  public KubernetesCredentials(List<String> namespaces, KubernetesClient client) {
     this.client = client;
-    this.namespace = namespace;
+    this.namespaces = namespaces;
   }
 
   public KubernetesClient getClient() {
     return client;
   }
 
-  public String getNamespace() {
-    return namespace;
-  }
+  public List<String> getNamespaces() { return namespaces; }
+
+  public Boolean isRegisteredNamespace(String namespace) { return namespaces.contains(namespace); }
```

Comment:
That's true - but it can't happen in the context of clouddriver - if `namespaces` is `null` in the clouddriver config, this object will have been instantiated with `namespaces = ["default"]`. I'll place a better exception here though just in case.